docs/feature_rawhid.md

```diff
@@ -0,0 +1,84 @@
+# Raw HID
+
+Raw HID allows for bidirectional communication between QMK and the host computer over an HID interface. This has many potential use cases, such as switching keymaps on the fly or changing RGB LED colors and modes.
+
+There are two main components to getting raw HID working with your keyboard.
+
+## Keyboard firmware
+
+The implementation is fairly straightforward for the firmware.
+In your `rules.mk` add:
+
+```make
+RAW_ENABLE = yes
+```
+
+In your `keymap.c` include `"raw_hid.h"` and implement the following:
+
+```C
+void raw_hid_receive(uint8_t *data, uint8_t length) {
+    // Your code goes here. data is the packet received from host.
+}
+```
+
+The `"raw_hid.h"` header also includes `void raw_hid_send(uint8_t *data, uint8_t length);`
+which allows sending packets from keyboard to host. It can also be used for debugging when building your host by returning all data back to host to test communication.
```

Comment:
```suggestion
which allows sending packets from keyboard to host. As an example, it can also be used for debugging when building your host application by returning all data back to the host.
```

Reply:
Should probably join the lines, too.

Reply:
Done